src/basic/xattr-util.c

```diff
@@ -104,7 +104,6 @@ static ssize_t getxattr_pinned_internal(
         if (n < 0)
                 return -errno;
 
-        assert((size_t) n <= size);
```

Comment:
Maybe better to move this to the caller, to avoid accessing outside of the allocated memory.